topology/probes/libvirt/libvirt.go

```diff
@@ -106,45 +131,122 @@ func (probe *Probe) getDomainInterfaces(
 		return
 	}
 	for _, itf := range d.Interfaces {
-		if constraint == "" || constraint == itf.Alias.Name {
+		if constraint == "" || (itf.Alias != nil && constraint == itf.Alias.Name) {
 			itf.Host = domainNode
 			itfObj := itf
 			interfaces = append(interfaces, &itfObj)
 		}
 	}
+	for _, hostdev := range d.HostDevices {
+		logging.GetLogger().Debugf("Found Hostdev %v", hostdev)
+		if hostdev.Mode != "subsystem" || hostdev.Type != "pci" {
+			continue
+		}
+		if hostdev.Source == nil || hostdev.Source.Address == nil {
+			continue
+		}
+		hostdev.Host = domainNode
+		hostdevs = append(hostdevs, &hostdev)
+	}
+	return
+}
+
+func (probe *Probe) makeHostDev(name string, source *Source) (node *graph.Node, err error) {
+	if source == nil || source.Address == nil {
+		err = errors.New("hostdev without PCI address")
+		return
+	}
+	address := formatPciAddress(source.Address)
+	node = probe.graph.LookupFirstChild(probe.root, graph.Metadata{
+		"BusInfo": address,
+	})
+	if node == nil {
+		m := graph.Metadata{
+			"Name":    fmt.Sprintf("%s-%s", name, strings.TrimPrefix(address, "0000:")),
```

Comment:
It could be nice to ass the Alias if present

Reply:
The problem is that libvirt creates some alias on the fly when the domain is really started. You have a name that changes... Note that we should fix the UI at some point as it does not reflect dynamically changing names (true also for interfaces).

Reply:
My point was to add the Alias if present not to use it as name, but we can address this in a following PR.

Reply:
makeHostDev is the core for HostDev nodes. Then as other libvirt interfaces they are enriched either by interface.ProcessNode if they were declared as interface or by enrichHostdev for raw host device. In both cases, if Alias is defined, it will be added. Did you find a path where it was not used ?

Reply:
my bad I didn't see the Alias thing in the enrichHostdev function. Thx